adapters/syslog/syslog.go

```diff
@@ -81,104 +158,80 @@ func NewSyslogAdapter(route *router.Route) (router.LogAdapter, error) {
 	if err != nil {
 		return nil, err
 	}
+	connIsUDP := reflect.TypeOf(conn).String() == "*net.UDPConn"
 
-	format := cfg.GetEnvDefault("SYSLOG_FORMAT", "rfc5424")
-	priority := cfg.GetEnvDefault("SYSLOG_PRIORITY", "{{.Priority}}")
-	pid := cfg.GetEnvDefault("SYSLOG_PID", "{{.Container.State.Pid}}")
-	hostname = getHostname()
-
-	tag := cfg.GetEnvDefault("SYSLOG_TAG", "{{.ContainerName}}"+route.Options["append_tag"])
-	structuredData := cfg.GetEnvDefault("SYSLOG_STRUCTURED_DATA", "")
-	if route.Options["structured_data"] != "" {
-		structuredData = route.Options["structured_data"]
+	var format Format
+	if format, err = getFormat(); err != nil {
+		return nil, err
 	}
-	data := cfg.GetEnvDefault("SYSLOG_DATA", "{{.Data}}")
-	timestamp := cfg.GetEnvDefault("SYSLOG_TIMESTAMP", "{{.Timestamp}}")
+	debug("setting format to:", format)
 
-	if structuredData == "" {
-		structuredData = "-"
-	} else {
-		structuredData = fmt.Sprintf("[%s]", structuredData)
+	var tmpl *FieldTemplates
+	if tmpl, err = getFieldTemplates(route); err != nil {
+		return nil, err
 	}
 
-	if isTCPConnecion(conn) {
-		if err = setTCPFraming(); err != nil {
+	var tcpFraming TCPFraming
+	if !connIsUDP {
```

Comment:
since `conn` is an interface, i don't think it's future safe to assume this logic will hold. this should probably be reverted to `isTCPConnecion`. to be clear, i'm totally fine with calling that once and setting as a value on the `Adapter` struct.

Reply:
Yes you are right. I also regret the poor naming of the `connIsUDP` boolean.
Will restore the `isTCPConnection()` function and fix the typo as well.